backend/api/v1/instance_service.go

```diff
@@ -976,25 +1001,51 @@ func convertToInstance(instance *store.InstanceMessage) (*v1pb.Instance, error)
 	}
 
 	return &v1pb.Instance{
-		Name:          fmt.Sprintf("%s%s", common.InstanceNamePrefix, instance.ResourceID),
+		Name:          buildInstanceName(instance.ResourceID),
 		Title:         instance.Title,
 		Engine:        engine,
 		EngineVersion: instance.EngineVersion,
 		ExternalLink:  instance.ExternalLink,
 		DataSources:   dataSourceList,
 		State:         convertDeletedToState(instance.Deleted),
-		Environment:   fmt.Sprintf("environments/%s", instance.EnvironmentID),
+		Environment:   buildEnvironmentName(instance.EnvironmentID),
 		Activation:    instance.Activation,
 		Options:       convertToInstanceOptions(instance.Options),
 		Roles:         convertToInstanceRoles(instance, instance.Metadata.GetRoles()),
 	}, nil
 }
 
+// buildRoleName builds the role name with the given instance ID and role name.
+func buildRoleName(b *strings.Builder, instanceID, roleName string) string {
+	b.Reset()
+	if _, err := b.WriteString(common.InstanceNamePrefix); err != nil {
+		return ""
+	}
```

Comment:
Let's `_, _ = b.WriteString` since it always returns nil error

Reply:
Yes, it is almost impossible to go wrong here unless there is a bigger error in the program.